infra/src/test/kotlin/fi/fta/geoviite/infra/ui/testgroup1/UserRoleTestUI.kt

```diff
@@ -68,30 +104,80 @@ fun assertPublicationLog(frontpage: E2EFrontPage) {
         .returnToFrontPage()
 }
 
-fun assertMapPage(role: E2ERole) {
+fun assertMapPage(role: E2ERole, trackNumber: TrackNumber, designName: String) {
     val mapPage = E2EAppBar().goToMap()
 
     when (role) {
         E2ERole.Operator,
         -> {
             mapPage
+                .also { assertDraftAndDesignModeTabsVisible() }
+                // TODO: Design mode is checked before draft mode because for whatever reason design mode selections fail
+                // if draft mode checks go before them. Find out why
+                .switchToDesignMode()
+                .also { it.toolBar.workspaceDropdown().selectByName(designName) }
+                .also { assertTrackLayoutPageEditButtonsVisible(it, trackNumber) }
                 .switchToDraftMode()
+                .also { assertTrackLayoutPageEditButtonsVisible(it, trackNumber) }
                 .goToPreview()
                 .waitForAllTableValidationsToComplete()
                 .goToTrackLayout()
                 .switchToOfficialMode()
         }
 
-        E2ERole.Team,
+        E2ERole.Team -> {
+            mapPage
+                .also { assertDraftAndDesignModeTabsVisible() }
+                .switchToDesignMode()
+                .also { it.toolBar.workspaceDropdown().selectByName(designName) }
+                .also { assertTrackLayoutPageEditButtonsInvisible(it, trackNumber) }
+                .switchToDraftMode()
+                .also { assertTrackLayoutPageEditButtonsInvisible(it, trackNumber) }
+        }
+
         E2ERole.Authority,
         E2ERole.Browser,
         E2ERole.Consultant,
         -> {
-            assertNull(mapPage.switchToDraftModeButton)
+            mapPage
+                .also { assertDraftAndDesignModeTabsInvisible() }
+                .also { assertTrackLayoutPageEditButtonsInvisible(it, trackNumber) }
         }
     }
 }
 
+fun assertDraftAndDesignModeTabsVisible() {
+    waitUntilExists(byQaId("draft-mode-tab"))
+    waitUntilExists(byQaId("design-mode-tab"))
+}
```

Comment:
Näitä tosiaan tuskin halutaan tarkastella milloinkaan muulloin, vaan halutaan vaan koittaa kliksutella asioita joiden pitäisi näkyä -> tein näistä testitiedoston sisäisiä funktioita, en `TrackLayoutPage`:n kamaa